plugin-catalog/src/components/plugins/List.tsx

```diff
@@ -0,0 +1,289 @@
+import {
+  Link as HeadlampRouterLink,
+  SectionHeader,
+} from '@kinvolk/headlamp-plugin/lib/CommonComponents';
+import {
+  Box,
+  Card,
+  CardActions,
+  CardContent,
+  CardMedia,
+  Divider,
+  Link,
+  Pagination,
+  TextField,
+  Tooltip,
+  Typography,
+} from '@mui/material';
+import { useEffect, useState } from 'react';
+import { Icon } from '@iconify/react';
+
+const PAGE_SIZE = 10;
+const ARTIFACTHUB_HEADLAMP_PLUGIN_KIND = '21';
+
+function fetchHeadlampPlugins(limit, sort, offset, ts_query) {
+  const url = 'https://artifacthub.io/api/v1/packages/search';
+
+  // Construct the query parameters
+  const params = {
+    kind: ARTIFACTHUB_HEADLAMP_PLUGIN_KIND,
+    facets: 'true',
+    limit: limit.toString(),
+    sort: sort.toString(),
+    offset: offset.toString(),
+    ts_query: ts_query.toString(),
+  };
+
+  // Convert query parameters to string
+  const queryString = new URLSearchParams(params).toString();
+
+  // Construct the final URL with query parameters
+  const finalUrl = `${url}?${queryString}`;
+
+  // Fetch the data
+  return fetch(`http://localhost:4466/externalproxy`, {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      'Forward-To': finalUrl,
+    },
+  })
+    .then(response => {
+      if (response.ok) {
+        return response.json();
+      }
+    })
+    .catch(error => {
+      throw new Error(`Request failed!${error}`);
+    });
+}
+
+function PluginCard(props) {
+  return (
+    <Box maxWidth="30%" width="400px" m={1}>
+      <Card>
+        <Box
+          heigh="60px"
+          display="flex"
+          alignItems="center"
+          marginTop="15px"
+          justifyContent="space-between"
+        >
+          <CardMedia
+            image={`https://artifacthub.io/image/${props.logo_image_id}`}
```

Comment:
Somehow this ends up being Grafana's icon.
You can default to this one: https://icon-sets.iconify.design/mdi/puzzle-outline/
We could make things more pleasant if we e.g. compute a color from the plugin's name (and then we can use some of the functions from the theme to ensure it's contrasting with the bg color).

Reply:
I have added the default image and the grafana icon displayed for the ai plugin because it is what is configured in my test repository. Regarding computing the color from plugins name I don't think it is straight forward to achieve if you have any ideas on how to achieve it let me know. IMHO this default icon is only shown only when the plugin isn't configured with a plugin, computing the color based on name is too much effort.